source/gui/settingsDialogs.py

```diff
@@ -278,9 +279,24 @@ def onSave(self):
 		"""
 		raise NotImplementedError
 
+	def preSave(self):

```

Comment:
I know I'm a little late to be reviewing this, however I would  much prefer this to be named like a question, e.g. `shouldSave` or `panelAllowsSave`

Reply:
Actually, this is essentially a validation step, so perhaps `isValid` would be better. At the moment the name indicates that it is called so that the panel can take actions before saving, rather than the importance of the return value.